deploy/helm/sumologic/templates/traces/otel-gateway/deployment.yaml

```diff
@@ -0,0 +1,113 @@
+      tolerations:
+        {{ toYaml .Values.otelgateway.deployment.tolerations | indent 8 }}
+      {{- end }}
+      {{- if .Values.otelgateway.deployment.priorityClassName }}
+      priorityClassName: {{ .Values.otelgateway.deployment.priorityClassName | quote }}
+      {{- end }}
+      # Otel agent quits if the load balancing backend (collector/samples) is
+      # not ready during first connect attempt. Restart policy `Always` guarantees that
+      # the agent will have a chance to retry when collector is ready.
+      restartPolicy: Always
+      containers:
+      - name: otelcontrib-gateway
+        image: {{ .Values.otelgateway.deployment.image.repository }}:{{ .Values.otelgateway.deployment.image.tag }}
+        imagePullPolicy: {{ .Values.otelgateway.deployment.image.pullPolicy }}
+        command:
+          - "/otelcontribcol"
+          - "--config=/conf/traces.otelgateway.conf.yaml"
+        env:
+        - name: GOGC
+          value: "80"
+{{- $ctx := .Values -}}
+{{- include "kubernetes.sources.envs" (dict "Context" $ctx "Type" "traces") -}}
+{{- include "kubernetes.sources.env" (dict "Context" $ctx "Type" "metrics" "Name" "default" ) | nindent 8 }}
+{{- if .Values.otelgateway.deployment.extraEnvVars }}
+{{- toYaml .Values.otelgateway.deployment.extraEnvVars | nindent 8 }}
+{{- end }}
+        resources:
+          {{- toYaml .Values.otelgateway.deployment.resources | nindent 10 }}
+        ports:
+        - containerPort: 5778  # Default endpoint for Jaeger Sampling.
+        - containerPort: 6831  # Default endpoint for Jaeger Thrift Compact receiver.
+          protocol: UDP
+        - containerPort: 6832  # Default endpoint for Jaeger Thrift Binary receiver.
+          protocol: UDP
+        - containerPort: 8888  # Default endpoint for querying metrics.
+        - containerPort: 9411  # Default endpoint for Zipkin receiver.
+        - containerPort: 14250 # Default endpoint for Jaeger gRPC receiver.
+        - containerPort: 14267 # Default endpoint for Jaeger TChannel receiver.
+        - containerPort: 14268 # Default endpoint for Jaeger HTTP receiver.
+        - containerPort: 55678 # Default endpoint for Opencensus receiver.
+        - containerPort: 4317  # Default endpoint for OTLP receiver.
+        - containerPort: 4318  # Default endpoint for OTLP HTTP receiver.
+        - containerPort: 55680 # Old endpoint for OTLP gRPC receiver.
+        - containerPort: 55681 # Default endpoint for OTLP HTTP receiver. (deprecated)
+        volumeMounts:
+        - name: otelgateway-config-vol
+          mountPath: /conf
+{{- if .Values.otelgateway.deployment.extraVolumeMounts }}
+{{- toYaml .Values.otelgateway.deployment.extraVolumeMounts | nindent 8 }}
+{{- end }}
+        livenessProbe:
+          httpGet:
+            path: /
+            port: 13133 # Health Check extension default port.
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 13133 # Health Check extension default port.
```

Comment:
Can you provide explicit configuration for these that is modifiable via values.yaml? At the very least, the livenessProbe should have a relatively high timeout, so it doesn't cause restarts under load.

Reply:
Done.